Move Remaining Styleguidist Stories to Storybook

SudharakaP opened the pull request:
Resolve https://github.com/opencollective/opencollective/issues/4313

Betree reviewed (commented):
I have a crash on https://oc-storybook-demo-pr-6585.onrender.com/?path=/story/contribution-flow-steppayment--default-story, otherwise what I tested looks good. Even if not perfect, with this migration our current Storybook is a nice first version that we should try to merge as soon as we can to finally get rid of styleguidist. Next steps after this:
- Redirect styleguide.opencollective.com
- Get rid of everything related to styleguidist in the code

SudharakaP commented:
> I have a crash on https://oc-storybook-demo-pr-6585.onrender.com/?path=/story/contribution-flow-steppayment--default-story, otherwise what I tested looks good. Even if not perfect, with this migration our current Storybook is a nice first version that we should try to merge as soon as we can to finally get rid of styleguidist. Next steps after this:
> 
>     * Redirect styleguide.opencollective.com
> 
>     * Get rid of everything related to styleguidist in the code

@Betree : I've just corrected the crash on that page. It seems that the `onChange` method handler was missing. Anyways, let me know if you see anything else or we could progress with the migration. 😄 

Let me investigate any missing pieces of styleguidiest to try to get rid of anything remaining. 😄

SudharakaP commented:
> Get rid of everything related to styleguidist in the code

@Betree : I've removed all references to Styleguidist and added some documentation on how to add a new storybook component as well. Hope it looks good now. 😄

Betree commented on docs/styleguide.md:
```diff
@@ -10,16 +10,62 @@ npm run styleguide:dev
 ## Create a new component:
 
-Only components with a matching example markdown file in the `styleguide/examples/` directory will appear in the styleguide. After creating a new component in the `components/` directory (i.e. `components/NewComponent.js`), make an example markdown file to go with it (i.e. `styleguide/examples/NewComponent.md`).
+Storybook files are defined in `mdx` format and placed in the `stories/` folder. Adding a new story could simply be done by 
```
Great job updating this documentation

SudharakaP replied:
Thanks, appreciated. 🤗 😄
